File: FindJobApp/src/main/java/com/qv_ct/pojos/Apply.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.qv_ct.pojos;

import java.io.Serializable;
import java.util.Date;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.Transient;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Size;
import org.springframework.web.multipart.MultipartFile;

/**
 *
 * @author nct68
 */
@Entity
@Table(name = "apply")
public class Apply implements Serializable{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Size(min = 5, max = 50, message = "{apply.title.sizeError}")
    private String title;
    @NotEmpty(message = "{apply.message.emptyError}")
    private String content;
    private String cv;
    @Column(name = "created_date")
    @Temporal(javax.persistence.TemporalType.DATE)
    private Date createdDate;
    private Boolean active = true;
    @ManyToOne
    @JoinColumn(name = "candidate_id")
    private User candidate;
    
    @ManyToOne
    @JoinColumn(name = "recruitment_id")
    private Recruitment recruitment;
    
    @Transient
    private MultipartFile file;

    public Apply() {
    }

    public Apply(String title, String content, User candidate, Recruitment recruitment) {
        this.title = title;
        this.content = content;
        this.candidate = candidate;
        this.recruitment = recruitment;
    }
    
    

    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the title
     */
    public String getTitle() {
        return title;
    }

    /**
     * @param title the title to set
     */
    public void setTitle(String title) {
        this.title = title;
    }

    /**
     * @return the content
     */
    public String getContent() {
        return content;
    }

    /**
     * @param content the content to set
     */
    public void setContent(String content) {
        this.content = content;
    }

    /**
     * @return the cv
     */
    public String getCv() {
        return cv;
    }

    /**
     * @param cv the cv to set
     */
    public void setCv(String cv) {
        this.cv = cv;
    }

    /**
     * @return the createdDate
     */
    public Date getCreatedDate() {
        return createdDate;
    }

    /**
     * @param createdDate the createdDate to set
     */
    public void setCreatedDate(Date createdDate) {
        this.createdDate = createdDate;
    }

    /**
     * @return the active
     */
    public Boolean getActive() {
        return active;
    }

    /**
     * @param active the active to set
     */
    public void setActive(Boolean active) {
        this.active = active;
    }

    /**
     * @return the candidate
     */
    public User getCandidate() {
        return candidate;
    }

    /**
     * @param candidate the candidate to set
     */
    public void setCandidate(User candidate) {
        this.candidate = candidate;
    }

    /**
     * @return the recruiter
     */
    public Recruitment getRecruitment() {
        return recruitment;
    }

    /**
     * @param recruitment the recruiter to set
     */
    public void setRecruitment(Recruitment recruitment) {
        this.recruitment = recruitment;
    }
    
    public MultipartFile getFile() {
        return file;
    }

    public void setFile(MultipartFile file) {
        this.file = file;
    }

    public void setFile(String file) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }
    
}

File: FindJobApp/src/main/webapp/resources/js/assets/apply.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


function enableApply(applyId, status) {
    if (status === 'active') {
        if (confirm("Ẩn thư ứng tuyển?") == true) {
            fetch(`/FindJobApp/api/admin/apply/inactive/${applyId}`, {
                method: "get",
                headers: {
                    "Content-Type": "application/json"
                }
            }).then(res => {
                if (res.status == 200) {
                    location.reload();
                } else
                    alert("Something wrong!!!");
            })
        }
    }

    if (status === 'inactive') {
        if (confirm("Kích hoạt lại thư ứng tuyển?") == true) {
            fetch(`/FindJobApp/api/admin/apply/active/${applyId}`, {
                method: "get",
                headers: {
                    "Content-Type": "application/json"
                }
            }).then(res => {
                if (res.status == 200) {
                    location.reload();
                } else
                    alert("Something wrong!!!");
            })
        }
    }
}

function deleteApply(applyId) {
    if (confirm("Xóa thư ứng tuyển?") == true) {
        fetch(`/FindJobApp/api/admin/apply/${applyId}`, {
            method: "delete",
            headers: {
                "Content-Type": "application/json"
            }
        }).then(res => {
            if (res.status == 200) {
                location.reload();
            } else
                alert("Something wrong!!!");
        })
    }
}